File: nysol_python/mcmd/methods/convTex/db/smp/mnjoin.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mnjoin'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
item,date,price
A,20081201,100
A,20081213,98
B,20081002,400
B,20081209,450
C,20081201,100
'''
db['idatas'].append(data)

data={}
data['name']='ref1.csv'
data['text']='''
item,cost
A,50
A,70
B,300
E,200
'''
db['idatas'].append(data)

data={}
data['name']='ref2.csv'
data['text']='''
item,cost
A,50
B,300
E,200
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
入力ファイルにある ``item`` 項目と、
参照ファイルにある ``item`` 項目を比較し同じ値の場合、 ``cost`` 項目を結合する。
入力ファイル、参照ファイル共に ``item=A`` が2行あり、結果、出力ファイルには2$\times$2=4行の ``item=A`` が出力されている。
'''
script['sh_code']='''
mnjoin k=item f=cost m=ref1.csv i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mnjoin(k="item", f="cost", m="ref1.csv", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='未結合データ出力'
script['comment']='''
``n=True`` を指定することで、参照ファイルにマッチしない入力ファイルの行( ``item="C"`` の行)も出力し、
``N=True`` を指定することで、入力ファイルにマッチしない参照ファイルの行( ``item="E"`` の行)も出力する。
'''
script['sh_code']='''
mnjoin k=item f=cost m=ref2.csv -n -N i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mnjoin(k="item", f="cost", m="ref2.csv", n=True, N=True, i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mnjoin.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
item,date,price
A,20081201,100
A,20081213,98
B,20081002,400
B,20081209,450
C,20081201,100
EOF
)}

File.open("ref1.csv","w"){|fpw| fpw.write(
<<'EOF'
item,cost
A,50
A,70
B,300
E,200
EOF
)}

File.open("ref2.csv","w"){|fpw| fpw.write(
<<'EOF'
item,cost
A,50
B,300
E,200
EOF
)}

############## Example 1
title="Basic Example "
comment=<<'EOF'
The \verb|item| field in the input file is compared with the \verb|item| field from the reference file, add \verb|cost| field for records with the same value. There are two records where \verb|item=A| in both input file and reference file, therefore, 2$\times$2=4 rows of \verb|item=A| is written to the output file.
EOF
scp=<<'EOF'
more dat1.csv
more ref1.csv
mnjoin k=item f=cost m=ref1.csv i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Ouput unmatched data"
comment=<<'EOF'
Use \verb|-n| to print records in the input data that do not match with those in the reference file (row where \verb|item="C"|), and use -N to print records in the reference file that do not match with those in the input file (row where \verb|item="E"|).
EOF
scp=<<'EOF'
more ref2.csv
mnjoin k=item f=cost m=ref2.csv -n -N i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)
